src/scene/graphics/shared/svg/SVGToGraphicsPath.ts

```diff
@@ -1,244 +1,202 @@
-                    data[dataIndex], data[dataIndex + 1],
+                    data[0], data[1],
                     lastX, lastY
                 );
-
-                dataIndex += 4;
                 break;
             case 'q':
-
                 path.quadraticCurveTo(
-                    lastX + data[dataIndex], lastY + data[dataIndex + 1],
-                    lastX + data[dataIndex + 2], lastY + data[dataIndex + 3]
+                    lastX + data[0], lastY + data[1],
+                    lastX + data[2], lastY + data[3]
                 );
 
-                lastX += data[dataIndex + 2];
-                lastY += data[dataIndex + 3];
-
-                dataIndex += 4;
+                lastX += data[2];
+                lastY += data[3];
                 break;
             case 'T':
-
-                lastX = data[dataIndex++];
-                lastY = data[dataIndex++];
+                lastX = data[0];
+                lastY = data[1];
 
                 path.quadraticCurveToShort(
                     lastX, lastY
                 );
-
                 break;
             case 't':
-
-                lastX += data[dataIndex++];
-                lastY += data[dataIndex++];
+                lastX += data[0];
+                lastY += data[1];
 
                 path.quadraticCurveToShort(
                     lastX, lastY
                 );
-
                 break;
             case 'A':
-
-                lastX = data[dataIndex + 5];
-                lastY = data[dataIndex + 6];
+                lastX = data[5];
+                lastY = data[6];
 
                 path.arcToSvg(
-                    data[dataIndex],
-                    data[dataIndex + 1],
-                    data[dataIndex + 2],
-                    data[dataIndex + 3],
-                    data[dataIndex + 4],
+                    data[0],
+                    data[1],
+                    data[2],
+                    data[3],
+                    data[4],
                     lastX, lastY
                 );
-
-                dataIndex += 7;
-
                 break;
             case 'a':
-                lastX += data[dataIndex + 5];
-                lastY += data[dataIndex + 6];
+                lastX += data[5];
+                lastY += data[6];
 
                 path.arcToSvg(
-                    data[dataIndex],
-                    data[dataIndex + 1],
-                    data[dataIndex + 2],
-                    data[dataIndex + 3],
-                    data[dataIndex + 4],
+                    data[0],
+                    data[1],
+                    data[2],
+                    data[3],
+                    data[4],
                     lastX, lastY
                 );
-
-                dataIndex += 7;
-
                 break;
             case 'Z':
             case 'z':
                 path.closePath();
+                if (subpaths.length > 0)
+                {
+                    currentSubPath = subpaths.pop();
```

Comment:
This will be faster code as ?? and ? usage generally do abunch of extra checks and thisway we only have one simple if check.

if(currentSubPath)
{
    lastX = currentSubPath.startX;
    lastY = currentSubPath.startY;
}
else
{
    lastX = 0;
    lastY = 0;
}

Reply:
Absolutely. Done.